pyomo/common/sorting.py

```diff
@@ -0,0 +1,133 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+class _robust_sort_keyfcn(object):
+    """Class for robustly generating sortable keys for arbitrary data.
+
+    Generates keys (for use with Python `sorted()` that are
+    (str(type_name), val), where val is the actual value (if the type
+    is comparable), otherwise the string representation of the value.
+    If str() also fails, we fall back on id().
+
+    This allows sorting lists with mixed types in Python3
```

Comment:
This is just a general comment - we only support Python3, so I don't know if it's worth calling out specifically.